code/processes/segmentedtickerplant.q

```diff
@@ -83,5 +83,38 @@ init:{
   .stplg.init[string .proc.procname];
  };
 
-// Have the init function called from torq.q
+// Loads the striping.json config file and logs any problems
+configload:{[scpath]
+     @[jsonchecks; 
+       .stpps.stripeconfig:@[{.j.k read1 x};scpath;{.lg.o[`configload;"Failed to load in striping.json file: ",x]}];
+       {.lg.o[`configload;"Datastriping is not enabled due to problems with striping.json"]}]
+     };
+
+// Checks if each subscriptiondefault is set for each segment and errors if not defined
+jsonchecks:{[.stpps.stripeconfig]
+     // check defaults are ignore or all
+     defaults:{first (flip .stpps.stripeconfig[x])[`subscriptiondefault]}each key .stpps.stripeconfig;
+     errors:1+ where {[x] not ("ignore"~x) or ("all"~x)}each defaults;
+     {if[0<count x;.lg.o[`jsonchecks;m:"subscriptiondefault not defined as \"ignore\" or \"all\" for segment ",string[x]," "]]}each errors;
```

Comment:
why iterate through them all? also I don't think `m` is used anywhere? and no need to have the space at the end of the log msg

```suggestion
     if[0<count errors;.lg.o[`jsonchecks;m:"subscriptiondefault not defined as \"ignore\" or \"all\" for segment(s) "," " sv string[errors]]];
```